pyproject.toml

```diff
@@ -54,6 +54,7 @@ mkdocs-material = "^7.3.6"
 mkdocs-git-revision-date-plugin = "^0.3.1"
 mike = "^0.6.0"
 mypy = "^0.931"
+types-dataclasses = "^0.6.4"
```

Comment:
could you explain why this is needed? If it's needed, are we lazy importing only during TYPE_CHECK phase?

Reply:
This package is needed by mypy to work correctly with python 3.6 due to:
```
Run make mypy
poetry run mypy --pretty aws_lambda_powertools
aws_lambda_powertools/utilities/idempotency/base.py:31:1: error: Library stubs
not installed for "dataclasses" (or incompatible with Python 3.6)  [import]
            import dataclasses
    ^
aws_lambda_powertools/utilities/idempotency/base.py:31:1: note: Hint: "python3 -m pip install types-dataclasses"
aws_lambda_powertools/utilities/idempotency/base.py:31:1: note: (or run "mypy --install-types" to install all missing stub packages)
aws_lambda_powertools/utilities/idempotency/base.py:31:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 125 source files)

make: *** [Makefile:91: mypy] Error 1
Error: Process completed with exit code 2.
```
Great idea with Lazy loading will add it.

Reply:
I see! Because it's Py3.6, it's EOL, and it's conditional import we can simply ignore the type `mypy.ini` 

```ini
[mypy-dataclasses]
ignore_missing_imports = True
```

Reply:
Ok, lets remove this package and ignore dataclassess errors